fix: bracket stripping in gemini responses

uinstinct opened the pull request:
## Description

Use streamSse instead of processGeminiResponse when receiving responses from Gemini streamChat .

resolves CON-4266
closes https://github.com/continuedev/continue/issues/2651

## AI Code Review

- **Team members only**: AI review runs automatically when PR is opened or marked ready for review
- Team members can also trigger a review by commenting `@continue-review`

## Checklist

- [] I've read the [contributing guide](https://github.com/continuedev/continue/blob/main/CONTRIBUTING.md)
- [] The relevant docs, if any, have been updated or created
- [] The relevant tests, if any, have been updated or created

## Screen recording or screenshot


https://github.com/user-attachments/assets/5bdc06f0-455f-41c1-97ef-dccdae2eabb8

https://github.com/user-attachments/assets/20589cfd-e968-4567-9170-ecb45f76acfc







## Tests

[ What tests were added or updated to ensure the changes work as expected? ]

    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Switched Gemini streamChat to use streamSse for proper SSE handling and direct chunk streaming, fixing broken/laggy responses and simplifying the streaming path.

- **Bug Fixes**
  - Iterate streamSse(response) and yield chunks directly.
  - Remove processGeminiResponse in the Gemini streaming flow.

<!-- End of auto-generated description by cubic. -->

RomneyDa reviewed (changes requested):
Something still feels off here, I think streamSse already does the buffer splitting which makes e.g. this line redundant:
`const parts = buffer.split("\n,");`

Thinking you should be able to remove streamResponse entirely from openai adapter and just use streamSse?

I think Vertex LLM class usage of processGeminiResponse would also require using streamSse. Alternatively, you could change processGeminiResponse to take response directly and use streamSse on it

uinstinct commented:
> Something still feels off here, I think streamSse already does the buffer splitting which makes e.g. this line redundant:
const parts = buffer.split("\n,");

it is needed for [parsing the gemini chat response](https://github.com/continuedev/continue/blob/31c7383e92913b2fee4b2603cde04394a9691921/core/llm/llms/Gemini.ts/#L323-L332)

> Thinking you should be able to remove streamResponse entirely from openai adapter and just use streamSse?

> I think Vertex LLM class usage of processGeminiResponse would also require using streamSse. Alternatively, you could change processGeminiResponse to take response directly and use streamSse on it

implemented!

RomneyDa commented:
> > Something still feels off here, I think streamSse already does the buffer splitting which makes e.g. this line redundant:
> > const parts = buffer.split("\n,");
> 
> it is needed for [parsing the gemini chat response](https://github.com/continuedev/continue/blob/31c7383e92913b2fee4b2603cde04394a9691921/core/llm/llms/Gemini.ts/#L323-L332)
> 

What I meant is I think streamSse already splits chunks by new lines so the split at "\n," is extraneous
https://github.com/continuedev/continue/blob/579bcbfcdda4dfe46d10d92f4b7eb329ecfeaebf/packages/fetch/src/stream.ts#L124-L150

RomneyDa commented on core/llm/llms/Gemini.ts:
```diff
@@ -317,15 +317,6 @@ class Gemini extends BaseLLM {
     let buffer = "";
     for await (const chunk of stream) {
```
why don't we just streamSse(stream) directly in processGeminiReponse?

sestinj commented:
:tada: This PR is included in version 1.3.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@continuedev/llm-info/v/1.3.0)
- [GitHub release](https://github.com/continuedev/continue/releases/tag/%40continuedev/llm-info%401.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

sestinj commented:
:tada: This PR is included in version 1.30.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@continuedev/config-yaml/v/1.30.0)
- [GitHub release](https://github.com/continuedev/continue/releases/tag/%40continuedev/config-yaml%401.30.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
